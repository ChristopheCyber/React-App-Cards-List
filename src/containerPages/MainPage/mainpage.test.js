import React from 'react';
// import { render } from '@testing-library/react';
import {MainPage} from './mainpage.component.jsx';

//ENZYME fncts: ( we don't use MOUNT and RENDER here )
import { shallow } from "enzyme";

/*
test('/Cards Selection Board/ text', () => {
  expect.assertions(1);
  const { getByText } = render(<MainPage />);
  const linkElement = getByText(/Cards Selection Board/i);
  expect(linkElement).toBeInTheDocument();
});
 */

const mockStore = {
  cards: [{
    "id": 1,
    "name": "React framework. ( card 1 )",
    "cardname": "React",
    "email": "ChrisCyb01@bla.biz",
    "address": {
      "street": "Kulas Light",
      "suite": "Apt. 556",
      "city": "Gwenborough",
      "zipcode": "92998-3874",
      "geo": {
        "lat": "-37.3159",
        "lng": "81.1496"
      }
    },
    "phone": "1-770-736-8031 x56442",
    "website": "https://reactjs.org/",
    "company": {
      "name": "Romaguera-Crona",
      "catchPhrase": "Multi-layered client-server neural-net",
      "bs": "e-business app web01"
    }
  }],
  searchTxt1: '',
  picContentType: "img-class",

  observable : () => null,
  dispatch : () => null,
  suscribe : () => null,
  getState : () => null,
  replaceReducer : () => null
};
describe('Group Script mainpage.test.I: ', () => {
  it('Test mainpage.test.I.1; Snapshot testing:', () => {
    expect.assertions(1);
    const wrapper = shallow( <MainPage tabCards={mockStore.cards} constSearchTxt1={mockStore.searchTxt1} constPicContentType={mockStore.picContentType} methodHandleChange={() => null}/> );
    // testing if shallow rendering correctly:
    expect(wrapper).toMatchSnapshot();
  });
});
/*
describe('Group Script mainpage.test.II: ', () => {
  it('Test mainpage.test.II.1; State testing:', () => {
    expect.assertions(1);
    //test MainPage :
    const wrapper = shallow( <MainPage/> );
    // testing :
    //*
    //.find() and simulate(events) methods comming with ENZYME:
    // wrapper.find('[id="searchBoxId"]').simulate('click');
    // wrapper.find('[id="searchBoxId"]').simulate('change');
    //*
    expect(wrapper.state('picContentType')).toEqual('img-class');
  });
});
*/
