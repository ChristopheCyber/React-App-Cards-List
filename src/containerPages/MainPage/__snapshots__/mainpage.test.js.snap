// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group Script mainpage.test.I:  Test mainpage.test.I.1; Snapshot testing: 1`] = `
<div
  className="MainPage img-class"
>
  <h1>
     Cards Selection Board 
  </h1>
  <label>
    <i
      className="bigLow js-grid"
    >
      Search in cards titles -&gt;
    </i>
  </label>
  <SearchBox
    handleChangeFct={[Function]}
    id="searchBoxId"
    propPlaceHolder="Type text for Searching"
  />
  <label>
    <i
      className="bigLow bigLow-search"
    >
      Â  
       ( 1 card in list now ) 
    </i>
  </label>
  <CardList
    cardsProp1={
      Array [
        Object {
          "address": Object {
            "city": "Gwenborough",
            "geo": Object {
              "lat": "-37.3159",
              "lng": "81.1496",
            },
            "street": "Kulas Light",
            "suite": "Apt. 556",
            "zipcode": "92998-3874",
          },
          "cardname": "React",
          "company": Object {
            "bs": "e-business app web01",
            "catchPhrase": "Multi-layered client-server neural-net",
            "name": "Romaguera-Crona",
          },
          "email": "ChrisCyb01@bla.biz",
          "id": 1,
          "name": "React framework. ( card 1 )",
          "phone": "1-770-736-8031 x56442",
          "website": "https://reactjs.org/",
        },
      ]
    }
    cardsPropImg="img-class"
  />
  <div
    id="tsparticles"
  >
    <h
      height="100%"
      id="tsparticles"
      options={
        Object {
          "detectRetina": true,
          "fpsLimit": 60,
          "interactivity": Object {
            "detectsOn": "window",
            "events": Object {
              "onClick": Object {
                "enable": true,
                "mode": "grab",
              },
              "onHover": Object {
                "enable": true,
                "mode": "push",
              },
              "resize": true,
            },
            "modes": Object {
              "bubble": Object {
                "distance": 400,
                "duration": 2,
                "opacity": 0.8,
                "size": 40,
              },
              "push": Object {
                "quantity": 4,
              },
              "repulse": Object {
                "distance": 50,
                "duration": 0.6,
              },
            },
          },
          "particles": Object {
            "collisions": Object {
              "enable": true,
            },
            "color": Object {
              "animation": Object {
                "enable": true,
                "speed": 40,
                "sync": true,
              },
              "value": "#ff0000",
            },
            "links": Object {
              "color": "#ffffff",
              "distance": 150,
              "enable": true,
              "opacity": 0,
              "width": 1,
            },
            "move": Object {
              "direction": "random",
              "enable": true,
              "outMode": "out",
              "random": true,
              "speed": 2,
              "straight": true,
            },
            "number": Object {
              "density": Object {
                "enable": true,
                "value_area": 1000,
              },
              "value": 80,
            },
            "opacity": Object {
              "value": 0.5,
            },
            "shape": Object {
              "type": "circle",
            },
            "size": Object {
              "random": true,
              "value": 7,
            },
          },
        }
      }
      style={Object {}}
      width="100%"
    />
  </div>
</div>
`;
